src/c-writer.cc

```diff
@@ -1633,7 +1633,7 @@ void CWriter::SerializeFuncType(const FuncType& func_type,
     *next_byte++ = MangleType(func_type.GetResultType(i));
   }
 
-  assert(next_byte - mangled_signature == len);
+  assert(next_byte - mangled_signature == static_cast<ptrdiff_t>(len));
```

Comment:
What is the error/warning that you see without this?

Reply:
`/src/src/c-writer.cc:1636:40: error: comparison of integers of different signs: 'long' and 'unsigned int' [-Werror,-Wsign-compare]`

because the result of `ptr-ptr` is `ptrdiff_t`

Reply:
Hmm.. I wonder why this is the only platform on which we see this?  Perhaps we don't build enough debug configurations?  I will investigate.

Reply:
Because the under layer type of ptrdiff_t is implementation defined. And in most GCC and Clang implementation, it's defined as `long` or `long int`.
When compiler got following code,
```
const char * a;
const char* b;
unsigned int x;
(a-b)==x;
```
It will do implicit type conversion
```
const char * a;
const char* b;
unsigned int x;
(a-b)==static_cast<long>(x);
```

And the key point there is, clang wasm build in our CI is 32bit platform, other builds are x86_64 or arm64.
In GCC/clang, `long` is 32 bit on 32bit platform and 64 bit on 64 bit platform.
Then on 64 bit platform, cast unsigned int to long is safe, but on 32 bit platform not. 
It's the reason why only emcc pipeline throw this error.

Reply:
Thanks.  I'm working on adding a debug i686 (32-bit x86) build so that this kind of thing won't happen again: #2369